features/apt_messages.feature

```diff
@@ -0,0 +1,328 @@
+        2 standard LTS security updates and 1 esm-apps security update
+
+        """
+
+        When I run `apt-get install -y --allow-downgrades <infra-pkg>` with sudo
+        When I run `apt-get install -y --allow-downgrades <apps-pkg>` with sudo
+        When I run `apt-get upgrade -y` with sudo
+        Then stdout matches regexp:
+        """
+        3 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+        2 esm-infra security updates and 1 esm-apps security update
+
+        """
+
+        When I run `apt-get install -y --allow-downgrades <standard-pkg>` with sudo
+        When I run `apt-get install -y --allow-downgrades <infra-pkg>` with sudo
+        When I run `apt-get install -y --allow-downgrades <apps-pkg>` with sudo
+        When I run `apt-get upgrade -y` with sudo
+        Then stdout matches regexp:
+        """
+        5 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+        2 standard LTS security updates, 2 esm-infra security updates and 1 esm-apps security update
+
+        """
+
+        When I run `apt-get upgrade -y` with sudo
+        Then stdout matches regexp:
+        """
+        0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+        """
+        Then stdout does not match regexp:
+        """
+        standard LTS security update
+        """
+        Then stdout does not match regexp:
+        """
+        esm-infra
+        """
+        Then stdout does not match regexp:
+        """
+        esm-apps
+        """
+
+        Examples: ubuntu release
+           | release | standard-pkg                                                          | infra-pkg                                            | apps-pkg |
+           | xenial  | accountsservice=0.6.40-2ubuntu10 libaccountsservice0=0.6.40-2ubuntu10 | curl=7.47.0-1ubuntu2 libcurl3-gnutls=7.47.0-1ubuntu2 | libzstd1=1.3.1+dfsg-1~ubuntu0.16.04.1 |
+
+    @series.xenial
+    @uses.config.machine_type.lxd.container
+    Scenario Outline: APT Hook advertises esm-infra on upgrade
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I run `apt-get update` with sudo
+        When I run `apt-get -y upgrade` with sudo
+        When I run `apt-get -y autoremove` with sudo
+        When I run `pro refresh messages` with sudo
+        When I run `apt-get upgrade` with sudo
+        Then stdout matches regexp:
+        """
+        Reading package lists...
+        Building dependency tree...
+        Reading state information...
+        Calculating upgrade...
+        The following security updates require Ubuntu Pro with 'esm-infra' enabled:
+          .*
+        Learn more about Ubuntu Pro for 16\.04 at https:\/\/ubuntu\.com\/16-04
+        0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded\.
+        """
+        When I attach `contract_token` with sudo
+        When I run shell command `apt-get upgrade || true` with sudo
```

Comment:
Why do we need the `|| true` here ?

Reply:
ah this was a quick hack to run apt-get upgrade without providing stdin and without actually upgrading.

I'll replace it by just using the --dry-run flag